File: Orbit-App/src/types/publicProfile.ts
import type {
  RecommendationReasonGroup,
  RecommendationScoreBreakdown,
} from "./recommendation";
import type { IntentMode } from "./profile";

export type PublicProfileCompatibility = {
  mutual_score: number;
  coverage_percentage: number;
  common_interests: string[];
  score_breakdown: RecommendationScoreBreakdown | null;
  reason_groups: RecommendationReasonGroup[];
};

export type PublicProfileRead = {
  profile_id: string;
  user_id: string;
  name: string;
  age: number | null;
  city: string | null;
  bio: string | null;
  photo_url: string | null;
  intent_mode: IntentMode;
  interests: string[];
  compatibility: PublicProfileCompatibility | null;
};

File: OrbitApi/app/schemas/public_profile.py
from uuid import UUID

from pydantic import BaseModel

from app.core.intent_mode_config import IntentMode
from app.schemas.recommendation import (
    RecommendationReasonGroupRead,
    RecommendationScoreBreakdownRead,
)


class PublicProfileCompatibilityRead(BaseModel):
    mutual_score: int
    coverage_percentage: int
    common_interests: list[str]
    score_breakdown: RecommendationScoreBreakdownRead | None
    reason_groups: list[RecommendationReasonGroupRead]


class PublicProfileRead(BaseModel):
    profile_id: UUID
    user_id: UUID
    name: str
    age: int | None
    city: str | None
    bio: str | None
    photo_url: str | None
    intent_mode: IntentMode
    interests: list[str]
    compatibility: PublicProfileCompatibilityRead | None
